pywbem/_listener.py

```diff
@@ -990,34 +1109,46 @@ def logger(self):
         """
         return self._logger
 
-    def stop_callback_thread(self):
+    @property
+    def max_ind_queue_size(self):
+        """
+        Return the maximum queue size. If this value is not zero, an
+        exception will be generated when this queue size is reached.
+        """
+        return self._max_ind_queue_size
+
+    def ind_delivery_queue_empty(self):
         """
-        Stop the indication delivery thread and queue and eliminate them. This
-        allows the delivery of indications already in the queue. and then
-        closes the thread and queue.
+        Return boolean True if the indication queue is empty. Otherwise return
+        False. This is available becasue the qsize attribute only returns an
+        approximation.
         """
-        assert self.callback_thread
-        assert self.rcvd_indication_queue
-        self.logger.info("Stopping queued delivery")
-        self.callback_thread.stop()
-        self.logger.info("Joining queued delivery")
-        self.callback_thread.join()
-        self.logger.info("Join finished threaded Deliver queue")
-        self.callback_thread = None
-        self.rcvd_indication_queue = None
+        return self.ind_delivery_queue.empty()
+
+    def qsize(self):
```

Comment:
Suggest to name this "queue_size()" since we use the full word "queue" on the other methods.

Reply:
Done